bulk_site_creator/views.py

```diff
@@ -99,14 +111,19 @@ def job_detail(request, job_id):
         'job': job,
         'tasks': tasks
     }
+    logger.debug(f'Retrieved job details for job {job_id}.', extra=context)
     return render(request, "bulk_site_creator/job_detail.html", context=context)
 
 
 @login_required
 @lti_role_required(const.ADMINISTRATOR)
 @lti_permission_required(settings.PERMISSION_BULK_SITE_CREATOR)
 @require_http_methods(["GET", "POST"])
-def new_job(request):
+def new_job(request: HttpRequest) -> HttpResponse:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This function calls 18 other functions. By comparison, 98% of the functions in the CodeGuru reference dataset use fewer. This indicates the function is highly coupled with other functions. A function that is highly coupled with other functions is difficult to understand and its behavior might change unexpectedly when one of its referenced functions is updated. High coupling could also increase the integration test complexity, maintenance cost and technical debt.

We recommend that you simplify this function or break it into multiple functions. For example, consider extracting the code block on lines 182-201 into a separate function.